upload-api/tables/index.js

```diff
@@ -26,3 +26,55 @@ export const uploadTableProps = {
   // space + root must be unique to satisfy index constraint
   primaryIndex: { partitionKey: 'space', sortKey: 'root' },
 }
+
+/** @type TableProps */
+export const delegationTableProps = {
+  fields: {
+    link: 'string',        // `baf...x`
+    audience: 'string',   // `did:web:service`
+    issuer: 'string',     // `did:key:agent`
+    expiration: 'string', // `9256939505` (unix timestamp)
+    insertedAt: 'string', // `2022-12-24T...`
+    updatedAt: 'string',  // `2022-12-24T...`
+  },
+  // TODO does this index setup seem right?
+  // we want to query by audience, but that won't necessarily be unique, so use cid as sortKey
+  primaryIndex: { partitionKey: 'audience', sortKey: 'link' },
+}
+
+/** @type TableProps */
+export const subscriptionTableProps = {
+  fields: {
+    cause: 'string',        // `baf...x` (CID of invocation that created this subscription)
+    provider: 'string',   // `did:web:service` (DID of the provider, e.g. a storage provider)
+    customer: 'string',   // `did:mailto:agent` (DID of the user account)
+    subscription: 'string',      // string (arbitrary string associated with this subscription)
```

Comment:
Mentioned it elsewhere, but I think we need two fields instead of one here:

1. `order` that MUST be unique per `provider`.
2. `id` (or `subscription`) which is derived from `order` & `provider` that is globally unique.

This would allow providers to enforce whatever limits they want and allow us to store records across all providers without collision risks.

Reply:
so I think I've managed to do this without introducing a new field - both tables are now indexed by `subscription` (primary key) and `provider` ("sort" key) which means that `subscription` will always be unique per-provider (since there will only ever be a single record for a given (`provider`, `subscription`)

I think this accomplishes what you're looking for without adding any user-space composite keys (as opposed to Dynamo composite keys, which is what we're creating with `subscription` and `provider`)